awswrangler/_data_types.py

```diff
@@ -725,6 +730,38 @@ def _cast_pandas_column(df: pd.DataFrame, col: str, current_type: str, desired_t
     return df
 
 
+def handle_oracle_decimal(con: Any, cursor_description: Any) -> Dict[str, pa.DataType]:
+    """Determine if a given Oracle column is a decimal, not just a standard float value."""
+    dtype = {}
+    if isinstance(con, oracledb.Connection):
```

Comment:
1. I hesitated a lot when putting them in `_data_types.py` (at first they were in `_databases.py`) - putting all Oracle-specific code in `oracle.py` makes more sense though - I've made the change
2. thank you! (forgot a bit about _oracle_found as it wasn't a good enough check when dealing with (3))
3. I've added it because I needed to convert float values to decimal only if they were from an Oracle database. Checking if `_oracle_found` was true was not enough (at it may be installed but not used if the current execution deals with postgres for example). It was a bit of a pain. However after your suggestions I think I can infer the engine without the con object, and apply the conversion only when relevant: if dtype forces a decimal type, and the column contains floats, then converting them is fair game. https://github.com/awslabs/aws-data-wrangler/commit/244f3b54a1416598f101e4f9b0d4d553d55bd718#diff-29379e00b37817ffc118369f8cf309f5cc3ffc6c16f6c547cb2f97af93afb587R454-R457